src/app.js

```diff
@@ -0,0 +1,60 @@
+let tasks = [];
+
+// Load tasks from local storage
+const storedTasks = localStorage.getItem('tasks');
+if (storedTasks) {
+  tasks = JSON.parse(storedTasks);
+}
+
+function saveTasksToLocalStorage() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+export function addTask(description) {
+  const newTask = {
+    description,
+    completed: false,
+    index: tasks.length + 1,
+    showDeleteIcon: false, // Set showDeleteIcon to false initially
+    showEdit: false,
+  };
+  tasks.push(newTask);
+  saveTasksToLocalStorage();
+}
+
+export function deleteTask(index) {
+  tasks = tasks.filter((task) => task.index !== index);
+  /* eslint-disable no-use-before-define */
+  updateIndexes();
+  saveTasksToLocalStorage();
+}
+
+export function editTask(index, newDescription) {
+  tasks[index].description = newDescription;
+  saveTasksToLocalStorage();
+}
```

Comment:
- If a user tries to edit a task, it doesn't seem to work as expected. on pressing the keyboard, nothing seem to be working. please, double check this behaviour.